boards/arm/qemu_cortex_m0/nrf_timer_timer.c

```diff
@@ -0,0 +1,188 @@
+/*
```

Comment:
Why did the whole driver have to be copied to the board directory?

Reply:
That's because there is no point having a second system clock driver based on Timer peripheral , besides the bed RTF driver. I only added the driver because the qemu does not emulate the RTC  :)

Reply:
Sorry, I'm confused. Where does the emulation come from then? I mean, it looks like nrf_timer_cc_read() is returning your counter value from somewhere, right? So the HAL is doing the emulation? If that's true, why won't the default driver work?

Reply:
@andyross the default nRF driver, used in Zephyr, is based on the NRF RTC peripheral. This peripheral is, however, not supported in the QEMU emulation of the Microbit platform (nRF51). Instead, the QEMU guys have added support for the NRF TIMER peripheral. So, I had to write a system timer driver, based on the NRF TIMER peripheral.

But there is no need to have this driver in the /drivers/timer directory as a generic timer driver; cause there is no use case for that; nRF SoCs use the nRF RTC timer that can utilize the low-frequency low-power clock source. This driver is not, even, specific to Microbit; microbit can use the default nRF RTC driver. This driver is specific for the qemu_cortex_m0 platform. Its sole purpose is to add system timer support for qemu cortex-m0, so we get coverage on ARMv6-M architecture.

Reply:
> nrf_timer_cc_read() is returning your counter value from somewhere, right? So the HAL is doing the emulation? If that's true, why won't the default driver work?

Correct. The default driver is a different peripheral. The "corresponding function, there, is called `nrf_rtc_cc_read()`. But as I said, this peripheral does not _yet_ work in qemu.